db/silkworm/db/access_layer.cpp

```diff
@@ -355,4 +356,77 @@ bool migration_happened(lmdb::Transaction& txn, const char* name) {
     return tbl->get(byte_view_of_c_str(name)).has_value();
 }
 
+std::optional<ChainConfig> read_chain_config(lmdb::Transaction& txn) {
+    using json = nlohmann::json;
+    auto genesis{read_block(txn, 0u, false)};
+    if (!genesis) {
+        return std::nullopt;
+    }
+
+    MDB_val config_key{kHashLength, static_cast<void*>(genesis->hash.bytes)};
+    auto config_value = txn.get(table::kConfig, &config_key);
+    if (!config_value) {
+        return std::nullopt;
+    }
+
+    /*
+    * Sample config
+    {
+   "byzantiumBlock":4370000,
+   "chainId":1,
+   "constantinopleBlock":7280000,
+   "daoForkBlock":1920000,
+   "daoForkSupport":true,
+   "eip150Block":2463000,
+   "eip150Hash":"0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0",
+   "eip155Block":2675000,
+   "eip158Block":2675000,
+   "ethash":{},
+   "homesteadBlock":1150000,
+   "istanbulBlock":9069000,
+   "muirGlacierBlock":9200000,
+   "petersburgBlock":7280000
+    }
+    */
+
+    auto config_json{json::parse(config_value->c_str())};
+    ChainConfig config{};
+
+    if (config_json.contains("chainId")) {
+        config.chain_id = config_json["chainId"].get<uint64_t>();
+    }
+    if (config_json.contains("homesteadBlock")) {
+        config.homestead_block.emplace(config_json["homesteadBlock"].get<uint64_t>());
```

Comment:
I'd simply say
```
config.homestead_block = config_json["homesteadBlock"].get<uint64_t>();
```
but that's a matter of taste.

Reply:
It throws if member does not exists

Reply:
Sorry I misread the comment ... yes maybe is a matter of taste. But as far as I understand emplace does not create a new std::optional and copies it to the member.

Reply:
I don't understand. How
```
config.homestead_block = config_json["homesteadBlock"].get<uint64_t>();
```
is different it terms of throwing from
```
config.homestead_block.emplace(config_json["homesteadBlock"].get<uint64_t>());
```
?

Reply:
Read my above. Sorry I posted it late

Reply:
No worries, both `emplace` and `=` are fine. I suspect the optimizer will produce the same code, but it's a minor thing.